engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/XenserverSnapshotStrategy.java

```diff
@@ -245,6 +245,12 @@ public boolean deleteSnapshot(Long snapshotId) {
             return true;
         }
 
+        if(snapshotVO.getState() == Snapshot.State.Allocated) {
```

Comment:
@anshul1886 the code to check and remove snapshot when they are stuck in Allocated state is already in line 223 above, in the same method. Why this redundancy, is this a synchronization issue?